### ISSUE TITLE
Manage deleted users/assets in the calculation of team/asset group scores

### ISSUE THREAD

**Issue by savacano28** [OpenBAS-Platform/openbas#1806]
## Reproducible Steps

1. Execute an inject with a team or assetgroup
2. Delete an user/asset 
3. Expectations linked to deleted users/assets are removed, but the score at the parent level remains unchanged

**Questions:** 
1. Should the user/asset be deleted or saved as archived (with no changes to the score)?
2. The same applies to teams and asset groups: should we keep teams/asset groups as deleted or archived?
3. Expectations for users/assets and teams/asset groups are not set in the same way. Should the child expectations reference the parent (team_id, assetgroupId)?

**Comment by RomuDeuxfois** [comment]
Need a global brainstorm to rework the TeamExpectation/playerExpectation and AssetGroupExpectation/AssetExpectation mechanism

**Comment by jborozco** [comment]
Agree, this bug is not the most prio, but we need to talk about our score logic to cover this use case
